### PR TITLE
prepare setuptools_scm based pipeline

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**
this is a initial pr, all it does is enable the tooling leaving out the bits where i don't have credentials

a possibly necessary discussion is how to deal with the changelog

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,30 +1,34 @@
...
-  - "2.6"
-  - "2.7"
-  - "pypy"
-  - "3.3"
-  - "3.4"
-  - "3.5"
-
+- '2.6'
+- '2.7'
+- pypy
```
Comment (immunda) [inline .travis.yml]: I'm assuming this is meant to be a string?

Comment (RonnyPfannschmidt) [inline .travis.yml]: the `travis setup pypi` command did re-serialize the file, and pypy is automatically a string

**Comment by davidism** [comment]
When I check out and `pip install -e .` this branch, I get a version like `2.2.dev63+ng71ab1b8.dev.20170226`, with the date on the end, even though the working dir is clean.

---

How do we indicate that the version should increment the 2nd value (2.**2**) when on the master branch, which represents work towards the next major release, versus incrementing the 3rd value (2.1.**1**) when on a maintenance branch for bugfixes?

In the manual process, I would bump master to 2.2 and branch off the last 2.1 tag for maintenance, bumping the branch to 2.1.1. When the branch merges back in to master, the 2.2 number is kept.

---

I want to sign the distributions before uploading them to PyPI. It doesn't look like Travis's PyPI deploy supports this.

---

Besides the changelog, we still have to manually bump the `__version__` attribute in the code.

**Comment by RonnyPfannschmidt** [comment]
@davidism a `2.2.0.dev` tag would signify a bump in a different component

if i forgot a `__version__` somewhere in code, thats my mistake, and i'd have to add it

if signatures are a must have, then currently there is no option to use travis, but it still removes a lot of churn

**Comment by RonnyPfannschmidt** [comment]
closing as i won`t be working on this pr anymore

**Pull request closed without merging**